src/server/repository/assessmentCycle/countryUserSummary/createOrReplaceView.ts

```diff
@@ -0,0 +1,64 @@
+import { Assessment, Cycle } from 'meta/assessment'
+
+import { BaseProtocol, DB, Schemas } from 'server/db'
+
+type Props = {
+  assessment: Assessment
+  cycle: Cycle
+  schemaName?: string
+}
+
+export const createOrReplaceView = (props: Props, client: BaseProtocol = DB) => {
+  const { assessment, cycle, schemaName = Schemas.getNameCycle(assessment, cycle) } = props
+  return client.query(
+    `
+        create or replace view ${schemaName}.country_user_summary as
+        with user_data as (
+            select
+                u.uuid as user_uuid,
+                u.email,
+                u.props->>'name' as name,
+                u.props->>'surname' as surname,
+                concat(
+                        coalesce(u.props->>'name', ''),
+                        case when u.props->>'name' is not null and u.props->>'surname' is not null then ' ' else '' end,
+                        coalesce(u.props->>'surname', '')
+                ) as fullname,
+                u.status
+            from users u
+        )
+        select distinct on (ud.user_uuid, coalesce(ur.country_iso, ui.country_iso))
+            ud.user_uuid,
+            coalesce(ur.country_iso, ui.country_iso)  as country_iso,
+            ud.name,
+            ud.surname,
+            ud.email,
+            ud.fullname,
+            ud.status,
+            coalesce(to_jsonb(ur.*), '{}'::jsonb)     as role,
+            case
+                when
+                  to_jsonb(ur.*) is null
+                then
+                  coalesce(to_jsonb(ui.*), '{}'::jsonb)
+                else
+                '{}'::jsonb
+            end                                     as invitation
```

Comment:
Fields are subject to change in upcoming PRs if needed (eg admin view needs some other data etc)